exec/java-exec/src/test/java/org/apache/drill/exec/DrillSeparatePlanningTest.java

```diff
@@ -0,0 +1,350 @@
+  //final String query = "SELECT o_orderkey FROM dfs.tmp.`multilevel` order by o_orderkey";
+  //final String query = "SELECT dir1, sum(o_totalprice) FROM dfs.tmp.`multilevel` where dir0 = 1995 group by dir1 order by dir1";
+  //final String query = String.format("SELECT dir0, sum(o_totalprice) FROM dfs_test.`%s/multilevel/json` group by dir0 order by dir0", TEST_RES_PATH);
+
+
+  @Test(timeout=30000)
+  public void testSingleFragmentQuery() throws Exception {
+    final String query = "SELECT * FROM cp.`employee.json` where  employee_id > 1 and  employee_id < 1000";
+
+    QueryPlanFragments planFragments = getFragmentsHelper(query);
+
+    assertNotNull(planFragments);
+
+    assertEquals(1, planFragments.getFragmentsCount());
+    assertTrue(planFragments.getFragments(0).getLeafFragment());
+
+    getResultsHelper(planFragments);
+  }
+
+  @Test(timeout=30000)
+  public void testMultiMinorFragmentSimpleQuery() throws Exception {
+    final String query = String.format("SELECT o_orderkey FROM dfs_test.`%s/multilevel/json`", TEST_RES_PATH);
+
+    QueryPlanFragments planFragments = getFragmentsHelper(query);
+
+    assertNotNull(planFragments);
+
+    assertTrue((planFragments.getFragmentsCount() > 1));
+
+    for ( PlanFragment planFragment : planFragments.getFragmentsList()) {
+      assertTrue(planFragment.getLeafFragment());
+    }
+
+    getResultsHelper(planFragments);
+  }
+
+  @Test(timeout=30000)
+  public void testMultiMinorFragmentComplexQuery() throws Exception {
+    final String query = String.format("SELECT dir0, sum(o_totalprice) FROM dfs_test.`%s/multilevel/json` group by dir0 order by dir0", TEST_RES_PATH);
+
+    QueryPlanFragments planFragments = getFragmentsHelper(query);
+
+    assertNotNull(planFragments);
+
+    assertTrue((planFragments.getFragmentsCount() > 1));
+
+    for ( PlanFragment planFragment : planFragments.getFragmentsList()) {
+      assertTrue(planFragment.getLeafFragment());
+    }
+
+    getResultsHelper(planFragments);
+
+  }
+
+  @Test(timeout=30000)
+  public void testPlanningNoSplit() throws Exception {
+    final String query = String.format("SELECT dir0, sum(o_totalprice) FROM dfs_test.`%s/multilevel/json` group by dir0 order by dir0", TEST_RES_PATH);
+
+    updateTestCluster(2, config);
+
+    List<QueryDataBatch> results = client.runQuery(QueryType.SQL, "alter session set `planner.slice_target`=1");
+    for(QueryDataBatch batch : results) {
+      batch.release();
+    }
+
+    DrillRpcFuture<QueryPlanFragments> queryFragmentsFutures = client.planQuery(QueryType.SQL, query, false);
+
+    final QueryPlanFragments planFragments = queryFragmentsFutures.get();
+
+    assertNotNull(planFragments);
+
+    assertTrue((planFragments.getFragmentsCount() > 1));
+
+    PlanFragment rootFragment = planFragments.getFragments(0);
+    assertFalse(rootFragment.getLeafFragment());
+
+    getCombinedResultsHelper(planFragments);
+
+    client.close();
```

Comment:
tests work both ways - I would not check in w/o successful tests runs :). But, yes, I am removing client.close() since it is done in base class @AfterClass  and in my tests I am anyway updating/resetting clients at the beginning of each test